lib/types.ts

```diff
@@ -55,8 +53,25 @@ export interface ScannedProjectCustom {
    * a future change should be mindful but find a way to unify them if possible.
    */
   depTree: DepTree | PkgTree;
-  targetFile?: string;
-  meta?: any;
+  targetFile?: string; // currently used for application-dependencies scans
+  meta?: any; // not to pollute with actual data; reserved for actual metadata
+}
+
+enum ScanType {
+  DependencyTree,
+  DependencyGraph,
+  ManifestFiles,
+}
+
+export interface ScannedProjectExtended extends ScannedProjectCustom {
+  scanType: ScanType;
+  // unknowingly structured data; determined by `scanType`
+  data: any;
```

Comment:
What about typing this as `unknown`? This disallows accessing properties of `data`, unless you assert its type or explicitly cast it (this way the consumer of the plugin response must actively check what they are receiving)

Reply:
sounds brilliant I think :)